rules/X++/ComplexityIndentationCombined.xq

```diff
@@ -0,0 +1,87 @@
+(: Combines Complexities and MethodIdentationLevel queries :)
+(: to Find methods on class or table types with a complexity level above 30 :)
+(: Error if maximum indentation of the method is above 2, warning otherwise :)
+(: @Language Xpp :)
+(: @Author pvillads@microsoft.com :)
+
+(: Simple rule of thumb: Count the branching and looping constructs and add 1. The if statements, for, 
+  while, and do/while constructs, each count as one. For the switch/case constructs, each case block counts as one. 
+  In if and ladder if constructs, the final else does not count. For switch/case constructs, the default block does not count. :)
+declare function local:MethodComplexity($m as element(Method)) as xs:integer
+{
+      1 + count($m//IfStatement) +  count($m//IfThenElseStatement) +  count($m//WhileStatement) + count($m//DoWhileStatement)
+      + count($m//ForStatement) + count($m/SearchStatement) + count($m//CaseValues/*) + count($m//ConditionalExpression)
+};
+
+(: The following functions are used to calculate the maximum indentation of a method :)
+declare function local:visitStatement($s as element()) as xs:integer
```

Comment:
can we call it maxIndentationOfStatement instead? The current name (that I originally came up with...) does not say much about what the method actually does. And then you would update the other names as shown below